pkg/storegateway/storepb/indexheader.proto

```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+syntax = "proto3";
+
+package thanos;
+
+option go_package = "storepb";
+
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+
+option (gogoproto.sizer_all) = true;
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+option (gogoproto.goproto_getters_all) = false;
+
+// Do not generate XXX fields to reduce memory footprint.
+option (gogoproto.goproto_unkeyed_all) = false;
+option (gogoproto.goproto_unrecognized_all) = false;
+option (gogoproto.goproto_sizecache_all) = false;
+
+message HeadersLazyLoadedTrackerState {
+  map<string, int64> lazy_loaded_blocks = 1;
+  string user_id = 2;
```

Comment:
How will this be used? Now we only persist the file in the sync directory of each tenant, do we also need the tenant ID in the proto file?

Reply:
I want to decouple tenant id from the location of the file. The cost is also very small just one field addition per tenant.